### ISSUE TITLE
No Userdefined External Uri Lookups

### ISSUE THREAD

**Issue by da2ce7** [torrust/torrust-index-gui#67]
Only admins should be able to create external uri lockups, for example, external hosted images.

All user defined uri lockups should be proxied through the back end.

**Comment by josecelano** [comment]
Hi @da2ce7, is this completed via the [image proxy with a cache in the backend](https://github.com/torrust/torrust-index-backend/tree/develop/src/web/api/v1/contexts/proxy)?

Or are you missing something else?

**Comment by josecelano** [comment]
@da2ce7 do you mean we should remove all links from the torrents description except for the PNG images served by our proxy? If yes, should we remove them in the frontend or in the backend (or both)?

Relates to: https://github.com/torrust/torrust-index-frontend/issues/105

Maybe we could replace the image URL with the proxied one and then use the [DOMPurify](https://github.com/cure53/DOMPurify) to clean the final HTML text (only in the frontend).

The package we are using to render the markdown ([marked](https://marked.js.org/#usage)) recommends the [DOMPurify](https://github.com/cure53/DOMPurify). By the way, it seems it has a [parser](https://marked.js.org/using_pro#parser), we could use it to change the image URls.

### Conclusion

- First at all, use the DOMPurify to remove dangerous HTML.
- Parse the text, removing all external links except for PNG images.
- Replace external PNG image links with the proxied linmk.

**Comment by da2ce7** [comment]
What I mean is that non-administrators should no be able to induce any external URI call for any other user.

We should go even so far to stop browsers from opportunistically pre-caching external links Proton-Mail style.

**Comment by josecelano** [comment]
Hi @da2ce7, I think I have also implemented this issue via the PR https://github.com/torrust/torrust-index-frontend/pull/158.

I've removed all external sources from HTML, so there are no external links to load (or prefetch). I've removed all HTML tags that are not converted directly from markdown. The only allowed tags with external links are: `a` and `img` and for anchor I've removed the "href", for images we are replacing the image with the proxied one.
